ffi/src/lib.rs

```diff
@@ -66,9 +126,20 @@ pub use utils::strings::{
 
 pub use utils::log;
 
+// type aliases for iterator types.
 pub type TypedValueIterator = vec::IntoIter<TypedValue>;
 pub type TypedValueListIterator = vec::IntoIter<Vec<TypedValue>>;
 
+/// A C representation of the a change provided by the transaction observers
+/// from a single transact.
+/// Holds a transaction identifier, the changes as a set of affected attributes
```

Comment:
Huh.  I understand (I think!) why the history points to exposing affected attributes -- that's what was needed for caching.  But why is this what we're exposing to consumers?

Datomic provides all the datoms transacted.  That's expensive for us, but maybe we should talk about why we don't do that and how to achieve it?  Maybe we should make it an option for transaction observers?  Maybe it's a special type of transaction observer that filters the affected attribute set and then pulls the relevant datoms?

Reply:
The reason we provide a list of the affected attributes is to enable consumers to decide which actions it wants to perform based on which attributes are affected. If I have an observer that is registered to listen to changes to `:foo/name` and `:foo/due_date`, I might want to do something different when if an `:foo/name` changes (just update the UI to reflect the change) to `:foo/due_date` (fire a notification to a user to let them know that the due date on an item has changed`. As we expand observers to further filter by entity id and acceptance criteria, this might become more important (name changed on entity a and due date changed on entity b in the same transaction:).